fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/TransactionImpl.java

```diff
@@ -101,7 +148,34 @@ public void commitIfShortLived() {
 
     @Override
     public void refresh() {
-        // TODO Auto-generated method stub
+        updateExpiry(timeout());
+    }
+
+    private Duration timeout() {
+        // TODO Get the user configured timeout?
+        // Otherwise, use the default timeout
+        return DEFAULT_TIMEOUT;
+    }
+
+    private PersistentStorageSession getPersistentSession() {
+        return this.txManager.getPersistentStorageSessionManager().getSession(this.id);
+    }
+
+    private void failIfExpired() {
+        if(hasExpired()) {
+            throw new RuntimeException("Transaction with transactionId: " + id + " expired!");
```

Comment:
We should at least throw RepositoryRuntimeExceptions for these methods. It would be better to throw more specific exceptions so we can set appropriate response codes. Are there exceptions leftover from the previous transaction implementation?